fuzz/build.rs

```diff
@@ -0,0 +1,138 @@
+fn main() -> anyhow::Result<()> {
+    #[cfg(feature = "component-model")]
+    component::generate_static_api_tests()?;
+
+    Ok(())
+}
+
+#[cfg(feature = "component-model")]
+mod component {
+    use anyhow::{anyhow, Context, Result};
+    use arbitrary::{Arbitrary, Unstructured};
+    use proc_macro2::TokenStream;
+    use quote::{format_ident, quote};
+    use rand::rngs::StdRng;
+    use rand::{Rng, SeedableRng};
+    use std::env;
+    use std::fmt::Write;
+    use std::fs;
+    use std::ops::DerefMut;
+    use std::path::PathBuf;
+    use std::process::Command;
+    use wasmtime_fuzzing::generators::component_types::{self, TestCase};
+
+    pub fn generate_static_api_tests() -> Result<()> {
+        println!("cargo:rerun-if-changed=build.rs");
+        let out_dir = PathBuf::from(
+            env::var_os("OUT_DIR").expect("The OUT_DIR environment variable must be set"),
+        );
+
+        let mut out = String::new();
+        write_static_api_tests(&mut out)?;
+
+        let output = out_dir.join("static_component_api.rs");
+        fs::write(&output, out)?;
+
+        drop(Command::new("rustfmt").arg(&output).status());
+
+        Ok(())
+    }
+
+    fn write_static_api_tests(out: &mut String) -> Result<()> {
+        let seed = if let Ok(seed) = env::var("WASMTIME_FUZZ_SEED") {
+            seed.parse::<u64>()
+                .with_context(|| anyhow!("expected u64 in WASMTIME_FUZZ_SEED"))?
+        } else {
+            StdRng::from_entropy().gen()
+        };
+
+        let mut rng = StdRng::seed_from_u64(seed);
+
+        const TEST_CASE_COUNT: usize = 100;
+
+        let mut tests = TokenStream::new();
+
+        let name_counter = &mut 0;
+
+        let mut declarations = TokenStream::new();
+
+        for index in 0..TEST_CASE_COUNT {
+            let mut bytes = vec![0u8; rng.gen_range(1000..2000)];
+            rng.fill(bytes.deref_mut());
+
+            let TestCase {
+                params,
+                result,
+                import_and_export,
+            } = TestCase::arbitrary(&mut Unstructured::new(&bytes))?;
+
```

Comment:
Could this perhaps be replaced with something that dynamically fills out more of the `bytes` vector when an error is encountered? Otherwise this could run into a possible issue where unrelated changes to the fuzz generator could cause this build to fail if just-the-wrong-seed is chosen.